api/tests/test_import_purchases.py

```diff
@@ -283,3 +283,17 @@ def test_encoding_autodetect_cp1252(self):
         self.assertEqual(Purchase.objects.count(), 1)
         self.assertEqual(Purchase.objects.first().description, "deuxième pomme")
         self.assertEqual(body["encoding"], "ISO-8859-1")
+
+    @authenticate
+    def test_fail_import_bad_format(self):
+        with open("./api/tests/files/bad_format_file.ods", "rb") as diag_file:
+            response = self.client.post(reverse("import_purchases"), {"file": diag_file})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        body = response.json()
+        errors = body["errors"]
+        first_error = errors.pop(0)
+        self.assertEqual(first_error["status"], 400)
+        self.assertEqual(
+            first_error["message"],
+            "Ce fichier est du format application/vnd.oasis.opendocument.spreadsheet, merci d'exporter votre fichier en format CSV et re-essayer.",
```

Comment:
J'en profiterais pour corriger le wording de ce message d'erreur (dans `api/views/diagnosticimport.py` & `api/tests/test_import_diagnostics`)
```suggestion
            "Ce fichier est au format application/vnd.oasis.opendocument.spreadsheet, merci d'exporter votre fichier au format CSV et ré-essayer.",
```

Reply:
corrigé ici : https://github.com/betagouv/ma-cantine/pull/4767/commits/7abb3225bc660618a225b33a5d94fb70cf34da4b